firebase-firestore/src/main/java/com/google/firebase/firestore/local/MemoryMutationQueue.java

```diff
@@ -69,14 +69,16 @@
   private ByteString lastStreamToken;
 
   private final MemoryPersistence persistence;
+  private final MemoryIndexManager indexManager;
 
-  MemoryMutationQueue(MemoryPersistence persistence) {
+  MemoryMutationQueue(MemoryPersistence persistence, MemoryIndexManager indexManager) {
     this.persistence = persistence;
     queue = new ArrayList<>();
 
     batchesByDocumentKey = new ImmutableSortedSet<>(emptyList(), DocumentReference.BY_KEY);
     nextBatchId = 1;
     lastStreamToken = WriteStream.EMPTY_STREAM_TOKEN;
+    this.indexManager = indexManager;
```

Comment:
The instance of IndexManager needs to be shared between all components, so that we share the same collection group index table. This is why I passed it in - but we do already use the same instance for all memory-backed implementations, regardless of user. It makes a ton of sense to just grab a reference here. 
It is semantically the same, but much more readable. Thanks for the suggestion.